arch/arm/include/inttypes.h

```diff
@@ -31,89 +31,130 @@
 
 #define PRId8       "d"
 #define PRId16      "d"
-#define PRId32      "ld"
+#if defined(__INT32_TYPE__) && __INT32_TYPE__ == 2
+#  define PRId32    "d"
+#else
+#  define PRId32    "ld"
+#endif
 #define PRId64      "lld"
-
 #define PRIdPTR     "d"
 
 #define PRIi8       "i"
 #define PRIi16      "i"
-#define PRIi32      "li"
+#if defined(__INT32_TYPE__) && __INT32_TYPE__ == 2
+#  define PRIi32    "i"
+#else
+#  define PRIi32    "li"
+#endif
 #define PRIi64      "lli"
-
 #define PRIiPTR     "i"
 
 #define PRIo8       "o"
 #define PRIo16      "o"
-#define PRIo32      "lo"
+#if defined(__INT32_TYPE__) && __INT32_TYPE__ == 2
+#  define PRIo32    "o"
+#else
+#  define PRIo32    "lo"
+#endif
 #define PRIo64      "llo"
-
 #define PRIoPTR     "o"
 
 #define PRIu8       "u"
 #define PRIu16      "u"
-#define PRIu32      "lu"
+#if defined(__INT32_TYPE__) && __INT32_TYPE__ == 2
+#  define PRIu32    "u"
+#else
+#  define PRIu32    "lu"
+#endif
 #define PRIu64      "llu"
-
 #define PRIuPTR     "u"
 
 #define PRIx8       "x"
 #define PRIx16      "x"
-#define PRIx32      "lx"
+#if defined(__INT32_TYPE__) && __INT32_TYPE__ == 2
+#  define PRIx32    "x"
+#else
+#  define PRIx32    "lx"
+#endif
 #define PRIx64      "llx"
-
 #define PRIxPTR     "x"
 
 #define PRIX8       "X"
 #define PRIX16      "X"
-#define PRIX32      "lX"
+#if defined(__INT32_TYPE__) && __INT32_TYPE__ == 2
```

Comment:
What about something like

```
#if defined(__INT32_TYPE__) && __INT32_TYPE__ == 2
#  define PRI32_PREFIX ""
#else
#  define PRI32_PREFIX "l"
#endif
```
And then use it in all places?

Reply:
You mean other arch? Since uint32_t definition in other arch is fixed, I think the current solution is fine.

Reply:
No. I mean reduce number of ifders in this file by:
```
#if defined(__INT32_TYPE__) && __INT32_TYPE__ == 2
# define PRI32_PREFIX ""
#else
# define PRI32_PREFIX "l"
#endif
...
#define PRId32 PRI32_PREFIX"d"
...
#define PRIX32 PRI32_PREFIX"X"
```

Reply:
Done.

Reply:
Since this approach need define more macro(e.g. STRINGIFY, CONCAT...), I prefer to keep the simple form.